File: 05题解-编程/javascript/sort/selectSort.ts
// ===
// Detail: sectct sort an array
// Example：[3, 7, 1, 9, 5] => [1, 3, 5, 7, 9]
// Tags: Array, sort
// Expertise: beginner
// ===

// Solution1:
// [3, 7, 1, 9, 5]
// for i 0 -> length-2
// min = i
// for j=i+1 -> length-1
// [min]> [j]  min =j
// end j loop
// min != i => exchange
function selectSort(arr: number[]): number[] {
  for (let i = 0; i < arr.length - 1; i++) {
    let min = i
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[min] > arr[j]) min = j
    }
    if (min !== i) [arr[min], arr[i]] = [arr[i], arr[min]]
  }

  return arr
}


// test
console.log(selectSort([3, 7, 1, 9, 5])) // => [ 1, 3, 5, 7, 9 ]

File: 图形化编程-scratch/简单数据结构和算法/select_sort.py

def select_sort(list):
    for i in range(0, len(list)):
        min = i
        for j in range(i+1, len(list)):
            if list[j] < list[min]:
                min = j
        if min != i:
            list[min], list[i] = list[i], list[min]


list = [1,7,9,5,3]
print(list)
select_sort(list)
print(list)